kivy/tests/test_window_base.py

```diff
@@ -18,3 +18,26 @@ def test_to_normalized_pos(self):
                 assert result_sy == expected_sy
         finally:
             win.system_size = old_system_size
+
+
+class WindowOpacityTest(GraphicUnitTest):
+
+    def test_window_opacity_property(self):
+        from kivy.core.window import Window
+
+        opacity = 0.5
+        Window.opacity = opacity
+        self.assertEqual(Window.opacity, opacity)
+
+        from kivy.logger import LoggerHistory
+
+        Window.opacity = 15
+        self.assertEqual(LoggerHistory.history[0].msg,
+                         'Window: The opacity value of '
+                         'the window should be in the '
+                         'range from 0.0 to 1.0.')
+
+        LoggerHistory.clear_history()
+
+        Window.opacity = -15
```

Comment:
Original `Window.opacity` should be stored/restored in `setUp`/`tearDown` methods. 

```python
def setUp(self):
    super().setUp()
    self._prev_window_opacity = self.Window.opacity

def tearDown(self):
    self.Window.opacity = self._prev_window_opacity
    super().tearDown()
```

Reply:
@mak8kammerer To answer your Discord question:

Methods `setUp`/`tearDown` should be added in `WindowOpacityTest` class.